cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -288,6 +288,11 @@ func WriteKubeletConfigFiles(cfg *kubeadmapi.InitConfiguration, patchesDir strin
 		errs = append(errs, errors.Wrap(err, "error writing kubelet configuration to file"))
 	}
 
+	// Writes the environment file with dynamic flags for the kubelet down to disk so the upgraded kubelet can start with fresh flags
+	if err := kubeletphase.WriteKubeletDynamicEnvFile(&cfg.ClusterConfiguration, &cfg.NodeRegistration, false, kubeletDir); err != nil {
+		return errors.Wrap(err, "error writing a dynamic environment file for the kubelet")
+	}
+
```

Comment:
this is already called in another place in upgrade, no?

what we need to do is open the existing env file on disk and remove any references of the --container-runtime-endpoint flag.